src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts

```diff
@@ -352,4 +346,52 @@ export class ViewConfigItemComponent implements OnInit, OnChanges, AfterViewChec
     }
   }
 
+  checkValidity(item: any = '', value = '') {
+    if (!isEmpty(this.categoryConfigurationData)) {
+      const data = this.categoryConfigurationData.value[0];
+      const config = [];
+      for (const k in data) {
+        if (data.hasOwnProperty(k)) {
+          data[k].value = data[k].value !== '' ? data[k].value : data[k].default;
+          if (item !== '' && k === item.key) {
+            data[k].value = value;
+          }
+          config.push(
+            {
+              key: k,
+              value: data[k].value
+            });
+        }
+      }
+
+      for (const k in data) {
+        if (data.hasOwnProperty(k)) {
+          data[k].key = k;
+          if (data[k].hasOwnProperty('validity')) {
+            data[k].eval = data[k].validity;
+            config.forEach(element => {
+              data[k].eval = data[k].eval.includes(element.key) ? data[k].eval
+                .replace(new RegExp(element.key, 'g'), `'${element.value}'`) : data[k].eval;
+            });
+          }
+        }
+      }
+
+      for (const k in data) {
+        if (data.hasOwnProperty(k)) {
+          if (data[k].hasOwnProperty('validity')) {
+            try {
```

Comment:
better to put try/catch thing inside if block `if (data[k]['validity'].trim() != "")`